generate/genhcl/genhcl_dynamic_test.go

```diff
@@ -315,83 +941,92 @@ func TestGenerateHCLDynamic(t *testing.T) {
+								labels("my_block", "nope"),
+								expr("for_each", `["a"]`),
 								content(
-									expr("value", "b.value"),
-									expr("key", "b.key"),
-									expr("other", "something.other"),
-									tmdynamic(
-										labels("child"),
-										expr("for_each", `[0, 1, 2]`),
-										expr("iterator", "i"),
-										content(
-											str("value", "${b.key}-${b.value}-${i.value}"),
-										),
-									),
+									str("a", "val"),
 								),
 							),
 						),
 					),
 				},
 			},
-			want: []result{
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:  "tm_dynamic with multiple content blocks fail",
+			stack: "/stack",
+			configs: []hclconfig{
 				{
-					name: "tm_dynamic_test.tf",
-					hcl: genHCL{
-						origin:    defaultCfg("/stack"),
-						condition: true,
-						body: hcldoc(
-							block("my_block",
-								number("key", 0),
-								expr("other", "something.other"),
-								str("value", "a"),
-								block("child",
-									str("value", "0-a-0"),
-								),
-								block("child",
-									str("value", "0-a-1"),
+					path: "/stack",
+					add: generateHCL(
+						labels("tm_dynamic_test.tf"),
+						content(
+							tmdynamic(
+								labels("my_block"),
+								expr("for_each", `["a"]`),
+								content(
+									str("a", "val"),
 								),
-								block("child",
-									str("value", "0-a-2"),
+								content(
+									str("a", "val"),
 								),
 							),
-							block("my_block",
-								number("key", 1),
-								expr("other", "something.other"),
-								str("value", "b"),
-								block("child",
-									str("value", "1-b-0"),
-								),
-								block("child",
-									str("value", "1-b-1"),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:  "tm_dynamic with unknown block fail",
+			stack: "/stack",
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: generateHCL(
+						labels("tm_dynamic_test.tf"),
+						content(
+							tmdynamic(
+								labels("my_block"),
+								expr("for_each", `["a"]`),
+								content(
+									str("a", "val"),
 								),
-								block("child",
-									str("value", "1-b-2"),
+								block("unsupported",
+									str("a", "val"),
 								),
 							),
-							block("my_block",
-								number("key", 2),
-								expr("other", "something.other"),
-								str("value", "c"),
-								block("child",
-									str("value", "2-c-0"),
-								),
-								block("child",
-									str("value", "2-c-1"),
-								),
-								block("child",
-									str("value", "2-c-2"),
+						),
+					),
+				},
+			},
+			wantErr: errors.E(hcl.ErrTerramateSchema),
+		},
+		{
+			name:  "tm_dynamic with undefined for_each fail",
+			stack: "/stack",
+			configs: []hclconfig{
+				{
+					path: "/stack",
+					add: generateHCL(
+						labels("tm_dynamic_test.tf"),
+						content(
+							tmdynamic(
+								labels("my_block"),
+								expr("for_each", `[local.a]`),
```

Comment:
Hmm interesting, seems possible. Using it as attributes could be tricky since we need proper strings/names as the attribute names, not partial expressions :thinking: